datafusion/sql/src/unparser/plan.rs

```diff
@@ -582,6 +582,7 @@ impl Unparser<'_> {
                     }
                     _ => (&join.left, &join.right),
                 };
+                let already_projected = select.already_projected();
```

Comment:
Could you add a comment explaining why we get `already_projected` before the rewriting for left_plan?